### ISSUE TITLE
service discovery: consul: fails to fetch all nodes

### ISSUE THREAD

**Issue by chen23** [F5Networks/f5-appsvcs-extension#59]
##### Do you already have an issue opened with F5 support?
No

##### ISSUE TYPE

 - Bug Report

##### AS3 BUILD/ VERSION

```
{
    "version": "3.8.1",
    "release": "1",
    "schemaCurrent": "3.8.0",
    "schemaMinimum": "3.0.0"
}
```

##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     13.1.1.2
  Build       0.0.4
  Edition     Point Release 2
  Date        Thu Oct 11 15:32:21 PDT 2018

```

##### SUMMARY

Not all nodes are being fetched from Consul

##### STEPS TO REPRODUCE

```
{
    "class": "ADC",
    "schemaVersion": "3.7.0",
    "id": "urn:uuid:33045210-3ab8-4636-9b2a-c98d22ab425d",
    "label": "Consul Service Discovery",
    "AS3Demo": {
      "class": "Tenant",
      "A1": {
        "class": "Application",
        "template": "http",
        "serviceMain": {
          "class": "Service_HTTP",
          "virtualAddresses": [
            "10.1.10.10"
          ],
          "pool": "web_pool"
        },
        "web_pool": {
          "class": "Pool",
          "monitors": [
            "http"
          ],
          "members": [
            {
              "servicePort": 8080,
              "addressDiscovery": "consul",
              "updateInterval": 10,
              "uri": "http://10.1.20.5:8500/v1/catalog/nodes",
              "encodedToken":"N2NhMTM3MWMtMzFmYS0xMWU4LThiYjItMDIyYzg4ZGFmODRl",
              "credentialUpdate": false
            }
          ]
        }
      }
    }
}
```

##### EXPECTED RESULTS

this should yield the following set of nodes (output from consul)
```
[
    {
        "ID": "",
        "Node": "10.1.1.6",
        "Address": "10.1.1.6",
        "Datacenter": "dc1",
        "TaggedAddresses": null,
        "Meta": null,
        "CreateIndex": 15,
        "ModifyIndex": 15
    },
    {
        "ID": "",
        "Node": "192.168.129.11",
        "Address": "192.168.129.11",
        "Datacenter": "dc1",
        "TaggedAddresses": null,
        "Meta": null,
        "CreateIndex": 8556,
        "ModifyIndex": 8556
    },
    {
        "ID": "",
        "Node": "192.168.129.12",
        "Address": "192.168.129.12",
        "Datacenter": "dc1",
        "TaggedAddresses": null,
        "Meta": null,
        "CreateIndex": 8557,
        "ModifyIndex": 8557
    },
    {
        "ID": "",
        "Node": "192.168.129.13",
        "Address": "192.168.129.13",
        "Datacenter": "dc1",
        "TaggedAddresses": null,
        "Meta": null,
        "CreateIndex": 8558,
        "ModifyIndex": 8558
    },
    {
        "ID": "f7621084-55a7-eff9-cf00-6e205dae5f71",
        "Node": "ip-10-1-1-6",
        "Address": "10.1.1.6",
        "Datacenter": "dc1",
        "TaggedAddresses": {
            "lan": "10.1.1.6",
            "wan": "10.1.1.6"
        },
        "Meta": {
            "consul-network-segment": ""
        },
        "CreateIndex": 5,
        "ModifyIndex": 6
    }
]
```

##### ACTUAL RESULTS

```
ltm pool A1/web_pool {
    members {
        /Common/consul--private:webcache {
            address 192.168.129.11
            monitor /Common/none
            session monitor-enabled
        }
        /Common/consul-f7621084-55a7-eff9-cf00-6e205dae5f71-private:webcache {
            address 10.1.1.6
            monitor /Common/none
            session monitor-enabled
        }
    }
    min-active-members 1
    monitor min 1 of { /Common/http }
    partition AS3Demo
}

```

**Comment by tpitkinf5** [comment]
Thanks @chen23 we are tracking this internally as 877

**Comment by chen23** [comment]
FYI I'd consider this low priority.  I really messed with the Consul API to create the nodes w/out an ID.  Ideally it should be safe to use the consul ID as a unique identifier.  After going back to create with an ID everything works as expected.

**Comment by kaustriaf5** [comment]
Fix slated for 3.10

**Comment by kaustriaf5** [comment]
Hi @chen23 this has been fixed in [3.10](https://github.com/F5Networks/f5-appsvcs-extension/releases/tag/v3.10.0).
